packages/material-ui/src/Menu/Menu.js

```diff
@@ -126,16 +200,70 @@ const Menu = React.forwardRef(function Menu(props, ref) {
       });
     }
 
+    // If the current item in this map has nestedItems,
+    // we need the Menu to orchestrate its nestedMenu
+    if (hasNestedMenu && parentMenuOpen) {
+      additionalPropsAdded = true;
+
+      const handleArrowRightKeydown = (event) => {
+        if (event.key === 'ArrowRight') {
+          event.preventDefault();
+          setLastEnteredItemIndex(index);
+        }
+      };
+      
+      Object.assign(additionalProps, {
+        handleArrowRightKeydown,
+        openNestedMenu: index === lastEnteredItemIndex && !entering,
+        setParentLastEnteredItemIndex: setLastEnteredItemIndex,
+      });
+    }
+
+    // If there are ANY children with nestedMenus, then ALL
+    // of the children need to know how to close any open nestedMenus
+    // and reset the state that controls which nested menu is open.
+    if (atLeastOneNestedMenu) {
+      additionalPropsAdded = true;
+
+
+      Object.assign(additionalProps, {
+        onMouseMove: (e) => {
+          setLastEnteredItemIndex(index);
+          if (onMouseMoveChildProp) {
+            onMouseMoveChildProp(e);
+          }
+        },
+        // parentMenuLevel: menuLevel,
+      });
+    }
+
+    // Using a semaphore instead of inspecting addtionalProps
+    // directly to avoid performance hits at scale. Might be
+    // fine to just do Object.keys(additionalProps).length > 0,
+    // but that seems like iterations we can avoid.
+    if (additionalPropsAdded) {
+      // eslint-disable-next-line consistent-return
+      return React.cloneElement(child, {
+        ...additionalProps,
+      });
+    }
+
+    // eslint-disable-next-line consistent-return
     return child;
   });
 
   return (
     <Popover
       getContentAnchorEl={getContentAnchorEl}
+      className={clsx({
+        [classes.disablePointerEvents]: nestedMenu
+      })}
       classes={PopoverClasses}
       onClose={onClose}
+      onEnter={handleEnter}
       onEntering={handleEntering}
-      anchorOrigin={theme.direction === 'rtl' ? RTL_ORIGIN : LTR_ORIGIN}
+      onEntered={handleEntered}
+      anchorOrigin={theme.direction !== 'rtl' ? RTL_ORIGIN : LTR_ORIGIN}
```

Comment:
whats the nature of this change?

Reply:
This needs to be reverted. This was a _super_ early change that I never went back and re-did correctly. The correct way to accomplish this is to make a similar change on the `React.cloneElement()` for the subMenu where we inject all those props.